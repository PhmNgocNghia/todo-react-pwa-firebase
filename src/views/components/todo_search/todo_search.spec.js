
import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import TodoSearch from './index';

describe('Todo Search', function() {
  // Render
  it('should render input when no text inside', function () {
    const shallowTodoSearch = shallow(<TodoSearch/>)
    expect(shallowTodoSearch.exists('input')).toBeTruthy()
  })

  // Event
  it('should call search todo when key press', function () {
    const mockUpdateFunc = jest.fn()
    const shallowTodoSearch = shallow(<TodoSearch onUpdateFilterKeywords={mockUpdateFunc} />)
    shallowTodoSearch.find('input').simulate('change', {target: {value: 'test'}})
    expect(mockUpdateFunc).toBeCalled()
  })


  // Snapshoot
  it('should match snapshoot', function () {
    const shallowTodoSearch = shallow(<TodoSearch/>)
    expect(toJson(shallowTodoSearch)).toMatchSnapshot()
  })
})