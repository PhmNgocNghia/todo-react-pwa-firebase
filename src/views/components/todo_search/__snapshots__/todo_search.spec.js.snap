// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Search should match snapshoot 1`] = `
<div>
  <styled.div
    className="position-relative"
  >
    <input
      className="form-control"
      id="todo_search_input"
      onChange={[Function]}
      placeholder="Hi, i'm a search box"
    />
  </styled.div>
</div>
`;
